odoo/addons/base/models/ir_rule.py

```diff
@@ -258,8 +258,8 @@ def get_record_description(rec):
             failing_rules = _("Blame the following rules:\n%s", rules_description)
             msg = f"{operation_error}\n{failing_records}\n\n{failing_rules}\n\n{resolution_info}"
 
-        # clean up the cache of records prefetched with display_name above
-        records_sudo.invalidate_recordset()
```

Comment:
Nitpick. You could rename `records_sudo` into `display_records` which is what it is used for.